go/trace_ctx.go

```diff
@@ -37,19 +37,26 @@ type TraceCtx struct {
 	telemetryId TelemetryId
 }
 
-func NewTraceCtx(adapter *AdapterBase, data []byte, config *Config) (*TraceCtx, error) {
+func NewTraceCtx(ctx context.Context, adapter *AdapterBase, r wazero.Runtime, data []byte, config *Config) (*TraceCtx, error) {
```

Comment:
if we don't anticipate (or want) this to be available to call outside our own package, call it `newTraceCtx` so its private.

Reply:
yeah we don't. i'm gonna go back and make as much stuff private as I can.